docs/spec/spec.md

```diff
@@ -0,0 +1,950 @@
+         * 5.1.2.2. [Service - LDAP User Store](#5122-service---ldap-user-store)
+         * 5.1.2.3. [Service - JWT](#5123-service---jwt)
+         * 5.1.2.4. [Service - OAuth2](#5124-service---oauth2)
+         * 5.1.2.5. [Client - Basic Auth](#5125-client---basic-auth)
+         * 5.1.2.6. [Client - Bearer Token Auth](#5126-client---bearer-token-auth)
+         * 5.1.2.7. [Client - Self Signed JWT Auth](#5127-client---self-signed-jwt-auth)
+         * 5.1.2.8. [Client - OAuth2](#5128-client---oauth2)
+6. [gRPC Utility Functions](#6-grpc-utility-functions)
+   * 6.1. [gRPC Deadline](#61-grpc-deadline)
+   * 6.2. [gRPC Compression](#62-grpc-compression)
+   * 6.3. [gRPC Access and Trace Logs](#63-grpc-access-and-trace-logs)
+   * 6.4. [gRPC Retry](#64-grpc-retry)
+
+
+# 1. Overview
+Ballerina gRPC standard library has five primary aspects in handling time values.
+1. gRPC CLI (command line interface)
+2. Protocol buffers to Ballerina data mapping
+3. gRPC communication
+4. gRPC Security
+5. gRPC utility functions
+
+# 2. gRPC Command Line Interface (CLI)
+
+Ballerina language comes up with a command-line interface that manages the lifecycle of a Ballerina program (such as build, test, and run). In addition, Ballerina CLI contains all the gRPC related stub and service skeleton generation capabilities. The gRPC command in Ballerina CLI is as follows.
+
+```sh
+bal grpc --input <proto-file-path> --output <output-directory> --mode client|service
+```
+
+The `--input` parameter is the only mandatory for the Ballerina gRPC command, and it specifies the path of the protobuf file of a gRPC service. The optional `--output` parameter indicates the path that output will be written. If the output path is not specified, the output will be written to a directory corresponding to the package in the Protocol Buffers definition. If the package is not specified, the output will be written to a `temp` directory in the current location. The optional `--mode` indicate what type of output files are needed. For example, if mode specifies as service, the gRPC command will generate the relevant stub file along with a service skeleton. If the mode is client, the gRPC command will generate a sample client code along with the stub. If nothing is specified, only the stub file is generated.
+
+# 3. Protocol Buffers to Ballerina Data Mapping
+
+The following table illustrates the data mapping of protocol buffers data types to relevant Ballerina types.
+
+|Protobuf Type|Ballerina Type|
+|---|---|
+|google.protobuf.DoubleValue|float|
+|google.protobuf.FloatValue|float|
+|google.protobuf.Int64Value|int|
+|google.protobuf.UInt64Value|int|
+|google.protobuf.Int32Value|int|
+|google.protobuf.UInt32Value|int|
+|google.protobuf.BoolValue|boolean|
+|google.protobuf.StringValue|string|
+|google.protobuf.BytesValue|byte[]|
+|google.protobuf.Empty|()|
+|google.protobuf.Timestamp|time:Utc|
+|google.protobuf.Duration|time:Seconds|
+|google.protobuf.Any|'any:Any|
```

Comment:
Need to add `struct`

Reply:
Fixed in here https://github.com/ballerina-platform/module-ballerina-grpc/pull/583/commits/543b1f0d2b2f7e6fafb4a9b3f987737cda8cc437